core/markets/retrieval/provider/retrieval_provider_impl.cpp

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <thread>
+
+#include "common/libp2p/cbor_stream.hpp"
+#include "markets/retrieval/provider/query_responder/query_responder_impl.hpp"
+#include "markets/retrieval/provider/retrieval_provider_impl.hpp"
+
+namespace fc::markets::retrieval::provider {
+  using common::libp2p::CborStream;
+
+  void RetrievalProviderImpl::start() {
+    host_service_->setProtocolHandler(
+        kQueryProtocolId, [self = shared_from_this()](StreamShPtr stream) {
+          self->query_responder_->onNewRequest(
+              std::make_shared<CborStream>(stream));
+        });
+    logger_->info("has been launched with ID "
+                  + host_service_->getPeerInfo().id.toBase58());
```

Comment:
You can use `peerInfoToPrettyString` from `"common/libp2p/peer/peer_info_helper.hpp"`

Reply:
getPeerInfo() returns PeerInfo structure with empty addresses, so `peerInfoToPrettyString` prints empty string, at current point will use ID for logging, in future will attempt to use suggested function)